community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -1142,15 +1146,52 @@ public Object graphGetProperty( KernelStatement state, int propertyKeyId )
     }
 
     @Override
-    public long countsForNode( KernelStatement statement, int labelId )
+    public long countsForNode( final KernelStatement statement, int labelId )
     {
-        return storeLayer.countsForNode( labelId );
+        long count = storeLayer.countsForNode( labelId );
+        if ( statement.hasTxStateWithChanges() )
+        {
+            CountsRecordState counts = new CountsRecordState();
+            try
+            {
+                statement.txState().accept( new TransactionCountingStateVisitor( null, storeLayer, this,
+                        statement, counts ) );
```

Comment:
Nice and simple solution. Visiting the state will of course do a bit more work than strictly needed, but less work than the previous solution which iterated through all nodes and counted...